src/celldega/qc/__init__.py

```diff
@@ -0,0 +1,161 @@
+
+        trx = trx.to_pandas()
+        trx = trx.rename(columns={'transformed_x': 'x_location', 'transformed_y': 'y_location', 'name': 'feature_name'})
+        trx_meta = trx[trx['cell_id'] != 'UNASSIGNED'][['transcript_id', 'cell_id', 'feature_name']]
+        
+        cell_gdf = get_cell_polygons(technology='Xenium', path_cell_boundaries=cell_boundaries_file, transformation_matrix=transform_file)
+        cell_gdf.reset_index(inplace=True)
+        cell_gdf['area'] = cell_gdf['geometry'].area
+        cell_gdf['centroid'] = cell_gdf['geometry'].centroid
+        cell_meta_gdf = cell_gdf[['cell_id', 'area', 'centroid']]
+        
+    percentage_of_assigned_transcripts = (len(trx_meta) / len(trx))
+
+    metrics['dataset_name'] = dataset_name
+    metrics['segmentation_approach'] = segmentation_approach
+    
+    metrics['proportion_assigned_transcripts'] = percentage_of_assigned_transcripts
+    metrics['number_cells'] = len(cell_gdf)
+    metrics['mean_cell_area'] = cell_gdf['geometry'].area.mean()
+    
+    metrics['mean_transcripts_per_cell'] = trx_meta.groupby(cell_index).size().mean()
+    metrics['median_transcripts_per_cell'] = trx_meta.groupby(cell_index)[transcript_index].count().median()
+
+    metrics['average_genes_per_cell'] = trx_meta.groupby(cell_index)[gene].nunique().mean()
+    metrics['median_genes_per_cell'] = trx_meta.groupby(cell_index)[gene].nunique().median()
+
+    metrics['proportion_empty_cells'] = ((len(cell_meta_gdf) - len(cell_gdf)) / len(cell_meta_gdf))
+
+    metrics_df = pd.DataFrame([metrics])
+    metrics_df = metrics_df.T
+    metrics_df.columns = [f"{dataset_name}-{segmentation_approach}"]
+    metrics_df = metrics_df.T
+    
+    gene_specific_metrics_df = pd.DataFrame({
+        "proportion_of_cells_expressing": (trx_meta.groupby(gene)[cell_index].nunique()) / len(cell_gdf),
+        "average_expression": (trx_meta.groupby(gene)[cell_index].nunique()) / (trx_meta.groupby(gene)[cell_index].nunique().sum()),
+        "assigned_transcripts": (trx_meta.groupby(gene)[transcript_index].count() / trx.groupby("feature_name")["transcript_id"].count()).fillna(0)
+    })
+
+    metrics_df.to_csv(path_output_cell_metrics)
+    gene_specific_metrics_df.to_csv(path_output_gene_metrics)
+
+    print("segmentation metrics calculation completed")
+
+def mixed_expression_calc(cell_feature_matrix_xenium_directory, cbg_stp_cellpose_default_file, cbg_stp_instanseg_file, cbg_stp_cellpose2_file, t_cell_specific_genes, b_cell_specific_genes):
+    
+    cbg_stp_cellpose2 = pd.read_parquet(cbg_stp_cellpose2_file)
+    cbg_stp_cellpose_default = pd.read_parquet(cbg_stp_cellpose_default_file)
+    cbg_stp_instanseg = pd.read_parquet(cbg_stp_instanseg_file)
+
+    cbg_xenium = read_cbg_mtx(cell_feature_matrix_xenium_directory)
+
+    cbg_dict = {}
+    cbg_dict['xenium_default'] = cbg_xenium
+    cbg_dict['cellpose_default'] = cbg_stp_cellpose_default
+    cbg_dict['instanseg'] = cbg_stp_instanseg
```

Comment:
remove references to specific segmentation algorithms and potentially set up something that takes a list/array of different methods to compare.